### PR TITLE
Add python-mpi4py package

### PR DESCRIPTION
**Pull request by ImperatorS79**
It compiles on my computer because I have Microsoft's msmpi SDK installed, I don't know what will happen on the CI.

### CODE REVIEW

File: mingw-w64-python-mpi4py/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+pkgdesc="Python bindings for the Message Passing Interface (MPI) standard (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+url='https://mpi4py.github.io/stable/'
+msys2_references=(
+  'pypi: mpi4py'
+)
+license=('BSD')
+depends=("${MINGW_PACKAGE_PREFIX}-python"
+         "${MINGW_PACKAGE_PREFIX}-msmpi")
```
Comment (mmuetzel) [inline mingw-w64-python-mpi4py/PKGBUILD:20]: The msmpi package from MSYS2 provides only the build dependencies (headers, link libraries, ...) for projects using MSMPI. It isn't actually a runtime dependency (and pulls in other heavy dependencies that aren't needed on runtime either).
Consider moving it to `makedepends`.

File: mingw-w64-python-mpi4py/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+pkgver=4.0.1
+pkgrel=1
+pkgdesc="Python bindings for the Message Passing Interface (MPI) standard (mingw-w64)"
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+url='https://mpi4py.github.io/stable/'
+msys2_references=(
+  'pypi: mpi4py'
+)
+license=('BSD')
```
Comment (mmuetzel) [inline mingw-w64-python-mpi4py/PKGBUILD:18]:
``` suggestion
license=('spdx:BSD-3-Clause')
```

(For a clickable link on the packages index.)

File: mingw-w64-python-mpi4py/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Maintainer: ImperatorS79 <fevrier.simon@gmail.com>
+
+_realname=mpi4py
+pkgbase=mingw-w64-python-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-python-${_realname}")
+provides=("${MINGW_PACKAGE_PREFIX}-python3-${_realname}")
+conflicts=("${MINGW_PACKAGE_PREFIX}-python3-${_realname}")
+replaces=("${MINGW_PACKAGE_PREFIX}-python3-${_realname}")
```
Comment (mmuetzel) [inline mingw-w64-python-mpi4py/PKGBUILD:8]: I might be missing something. But was there ever a package called `python3-mpi4py` that this new package should replace?
Should these lines just be removed?

Comment (ImperatorS79) [reply mingw-w64-python-mpi4py/PKGBUILD:8]: I copied the PKGBUILD from another python package, that is why those lines are here. I will remove them.

**Comment by mmuetzel** [comment]
LGTM now. 👍

Thank you for your contribution.

**Pull request merged**